lib/rubocop/cop/lint/ambiguous_block_association.rb

```diff
@@ -81,7 +81,7 @@ def ambiguous_block_association?(send_node)
         def allowed_method_pattern?(node)
           node.assignment? || node.operator_method? || node.method?(:[]) ||
             allowed_method?(node.last_argument.method_name) ||
```

Comment:
How about the same problem with `AllowedMethods`?

Reply:
I think there it's fine to match on the name. Generally, it's a bit unclear which patterns would need to match on the entire `send` node's source.

Reply:
`AllowedMethods` and `AllowedPatterns ` have different uses, so `AllowedMethods` seems fine.